packages/core/strapi/lib/services/cron.js

```diff
@@ -44,9 +44,13 @@ const createCronService = () => {
     },
     remove(name) {
       if (!name) throw new Error('You must provide a name to remove a cron job.');
-      const matchingJobsSpecs = jobsSpecs.filter(({ name: jobSpecName }) => jobSpecName === name);
+      const matchingJobsSpecs = jobsSpecs.filter(({ name: jobSpecName }, index) => {
+        if (jobSpecName === name) {
+          jobsSpecs.splice(index, 1);
+          return true;
+        }
+      });
```

Comment:
```suggestion
          return true;
        }
        return false;
      });
```

Needs an explicit return value for lint